tubemq-client/src/main/java/org/apache/tubemq/client/producer/SimpleMessageProducer.java

```diff
@@ -321,7 +322,7 @@ private void checkMessageAndStatus(final Message message) throws TubeClientExcep
         builder.setTopicName(partition.getTopic());
         builder.setPartitionId(partition.getPartitionId());
         builder.setData(ByteString.copyFrom(encodePayload(message)));
-        builder.setFlag(MessageFlagUtils.getFlag(message));
+        builder.setFlag(MessageFlagUtils.getFlag(message, true));
```

Comment:
The function of payload data process is recommended to be placed on line 321, and the attribute need to be adjusted by the function, 
 and there is no need to add a mandatory flag setting, which will make the maintenance personnel confused later.